### PR TITLE
Some change to repeated_solver.py

### PR DESCRIPTION
**Pull request by XiangyuG**

### CODE REVIEW

File: sol_verify.py [hunk 1]
```diff
@@ -39,8 +40,38 @@
...
-    sys.exit(1)
+  #failed
+  print("Sketch failed.")
+  sys.exit(1)
 else:
-    print("Sketch succeeded.")
-    sys.exit(0)
+  print("Sketch succeeded.")
+  sys.exit(0)
+"""
```
Comment (taegyunkim) [inline sol_verify.py:49]: Why is this in a comment  block? If it's unnecessary just delete it. Git lets us to revert if necessary.

File: sol_verify.py [hunk 2]
```diff
@@ -39,8 +40,38 @@
...
+  sketch_file_with_hole_value = hole_value_file_string + original_sketch_file_string
+
+  # Create file and write sketch_harness into it.
+  sketch_file = open(original_sketch_file[0:original_sketch_file.find('.')] + "_with_hole_value.sk", "w")
+  sketch_file.write(sketch_file_with_hole_value)
+  sketch_file.close()
+
+  # Call sketch on it
+  (ret_code, output) = subprocess.getstatusoutput("sketch --bnd-inbits=10 " + sketch_file.name)
+  if (ret_code!=0):
```
Comment (taegyunkim) [inline sol_verify.py:73]: Simply return ret_code here,
```Python
return ret_code
```

File: sol_verify.py [hunk 3]
```diff
@@ -39,8 +40,38 @@
...
 else:
-    print("Sketch succeeded.")
-    sys.exit(0)
+  print("Sketch succeeded.")
+  sys.exit(0)
+"""
+def sol_verify(original_sketch_file,hole_value_file):
+
+  original_sketch_file_string    = open(str(original_sketch_file),"r").read()
+  hole_value_file_string         = open(str(hole_value_file),"r").read()
```
Comment (taegyunkim) [inline sol_verify.py:53]: After opening a file, please call `close()` on it. Or do the following,
```Python
from pathlib import Path

original_sketch_file_string = Path(str(original_sketch_file)).read_text() 
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text

Or you can use `with` statement as in here 
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

File: sol_verify.py [hunk 4]
```diff
@@ -39,8 +40,38 @@
...
+
+  #remove the redundant code in hole_value_file
+  begin_pos = hole_value_file_string.find('int')
+  end_pos = hole_value_file_string.rfind(';')
+  hole_value_file_string = hole_value_file_string[begin_pos:end_pos+1]
+
+  sketch_file_with_hole_value = hole_value_file_string + original_sketch_file_string
+
+  # Create file and write sketch_harness into it.
+  sketch_file = open(original_sketch_file[0:original_sketch_file.find('.')] + "_with_hole_value.sk", "w")
```
Comment (taegyunkim) [inline sol_verify.py:67]: Here you closed the file, but still it might be less redundant to use `Path` or `with` statement,

**Comment by taegyunkim** [comment]
Hey Xiangyu,

Thanks for the merge, but I also left some comments there. Please take a
look and ask me questions if anything's unclear.

https://github.com/anirudhSK/chipmunk/pull/23/files

File: sol_verify.py (hunk 1 above)
Comment (XiangyuG) [reply sol_verify.py:49]: This part is just for testing. I do not want to delete it right now but we can delete it before the final version

File: sol_verify.py (hunk 3 above)
Comment (XiangyuG) [reply sol_verify.py:53]: Yeah. I think this is a good suggestion, I will fix this part

**Pull request merged**